src/hooks/useRequestProcessor.js

```diff
@@ -0,0 +1,66 @@
+import { useCallback } from 'react'
+import { useHeartbeat } from './useCourtContracts'
+import { useCourtClock } from '../providers/CourtClock'
+import { useRequestQueue } from '../providers/RequestQueue'
+import { useActivity } from '../components/Activity/ActivityProvider'
+
+import radspec from '../radspec'
+import allowedTermsBehind from '../allowedTermsBehind'
+
+export function useRequestProcessor() {
+  const { request } = useHeartbeat()
+  const { addActivity } = useActivity()
+  const { addRequests } = useRequestQueue()
+  const { neededTransitions } = useCourtClock()
+
+  const processRequests = useCallback(
+    requests => {
+      const processedRequests = []
+        .concat(requests)
+        .map(
+          ({
+            action,
+            description,
+            ensureConfirmation = false,
+            isTx = true,
+            name,
+            params,
+            ...request
+          }) => {
+            const intent = isTx
+              ? () => addActivity(action(), name, params)
+              : action
+
+            return {
+              name,
+              intent,
+              description:
```

Comment:
It might just be me, but I actually like being more explicit about the radspec description and passing it down from the request creation site.

I think that would allow us to avoid passing down the params (which seems like an odd thing to do if we can avoid it—does `addActivity()` really need it or can it also just use the passed down description?)

Reply:
Good point, i think the way you suggest results in a more abstracted functionality.

> does addActivity() really need it or can it also just use the passed down description?)

It does need it yes, but we can change it to pass down the full description. Besides the params passed are actually used to compute the description using `radpsec` again when getting the activity data, si we definitely need to update that.

Reply:
Since we will be  wiping older activities, a backwards compat implementation is not needed (see https://github.com/aragon/court-dashboard/pull/341)